brd-android/app/src/main/java/com/breadwallet/ui/usdaccount/transactions/UsdAccountTransactionsController.kt

```diff
@@ -56,17 +74,26 @@ class UsdAccountTransactionsController(args: Bundle? = null) : BaseMobiusControl
     override fun M.render() {
         with(binding) {
             ifChanged(M::transactions) {
-                rvTransactions.adapter = UsdAccountTransactionsAdapter(transactions)
+                if (it.size <= TRANSACTION_PAGE_LIMIT) {
+                    rvTransactions.adapter = UsdAccountTransactionsAdapter(transactions)
+                } else {
+                    (rvTransactions.adapter as UsdAccountTransactionsAdapter).updateData(it)
+                }
```

Comment:
Create a variable for adapter and connect it with RecyclerView in onCreateView function. Here we just need to update data

```suggestion
                if (it.size <= TRANSACTION_PAGE_LIMIT) {
                    adapter.setData(transactions)
                } else {
                    adapter.updateData(transactions)
                }
```